src/components/form/messages_spec.ts

```diff
@@ -1,93 +1,134 @@
           expect(api.container.isTouched).toBe(false);
           api.fixture.destroy();
         });
       });
 
-      it('should bind local view references #ref="ngForm"', () => {
-        return setup().then((api: IFormMessagesFixture) => {
+      it('should bind local view references formControlName="ref"', () => {
+        return setup().then((api:IFormMessagesFixture) => {
           expect(api.container.isTouched).toBe(false);
           expect(api.messages.length).toBe(1);
-          expect(api.container.form).not.toBeNull();
-          expect(api.fixture.componentInstance.name).toBe('MorTon');
+          expect(api.container.formDirective).not.toBeNull();
+          expect(api.container.control.value).toBe('MorTon');
         });
       });
 
-      it('should re-export valid from control or form', async(() => {
-        return setup().then((api: IFormMessagesFixture) => {
-          return promiseWait().then(() => {
-            let ctrl: Control = (<any>api.container.property).control;
-            expect(ctrl.value).toBe(null);
-            expect(api.container.valid).toBe(false);
-            expect(api.container.valid).toBe(ctrl.valid);
-            ctrl.updateValue('MorTon', {emitEvent: true});
-            api.fixture.detectChanges();
-            expect(api.container.valid).toBe(true);
-          });
+      it('should add md-valid and md-invalid class on [mdMessages] based on field validation state', () => {
+        return setup().then((api:IFormMessagesFixture) => {
+          expect(api.fixture.debugElement.query(By.css('[md-messages].md-valid'))).toBeNull();
+          expect(api.fixture.debugElement.query(By.css('[md-messages].md-invalid'))).toBeNull();
+          api.container.control.updateValue(null, {emitEvent: true});
+          let input = api.fixture.debugElement.query(By.css('input'));
+          input.nativeElement.focus();
+          input.nativeElement.blur();
+          api.fixture.detectChanges();
+          expect(api.fixture.debugElement.query(By.css('[md-messages].md-invalid'))).not.toBeNull();
+          api.container.control.updateValue('MorTon', {emitEvent: true});
+          input.nativeElement.focus();
+          input.nativeElement.blur();
+          api.fixture.detectChanges();
+          expect(api.fixture.debugElement.query(By.css('[md-messages].md-invalid'))).toBeNull();
+          expect(api.fixture.debugElement.query(By.css('[md-messages].md-valid'))).not.toBeNull();
         });
-      }));
+      });
+
+      it('should re-export valid from control or form', () => {
+        return setup().then((api:IFormMessagesFixture) => {
+          api.container.control.updateValue(null, {emitEvent: true});
+          api.fixture.detectChanges();
+          expect(api.container.control.value).toBe(null);
+          expect(api.container.valid).toBe(false);
+          expect(api.container.formDirective.valid).toBe(false);
+          api.container.control.updateValue('MorTon', {emitEvent: true});
```

Comment:
One it per test. This test is build to detect change so it shouldn't test the default state. If you want to test default state, it should be a test only for this